remove AWS authentication params from optional params

5S opened the pull request:
## Title

remove AWS authentication params from optional params to prevent completion call failures

## Relevant issues

Related to #9106

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the `tests/litellm/` directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on (`make test-unit`)[https://docs.litellm.ai/docs/extras/contributing_code]
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

## Type

🐛 Bug Fix

## Changes

- remove AWS authentication params from optional params in `litellm/llms/bedrock/chat/invoke_transformations/anthropic_claude3_transformation.py`.

krrishdholakia commented:
Hi @5S i see no test cases added where failures would be triggered. 

We do not want to pop the value in all places, as it might be needed at different points.

5S committed:
fix(bedrock): remove AWS authentication params from optional params to prevent completion call failures

5S commented:
Hi @krrishdholakia, thank you for your feedback.
I've updated the PR with a more focused scope and added test cases.
Please let me know if you have any concerns.

krrishdholakia commented on litellm/llms/bedrock/chat/invoke_transformations/anthropic_claude3_transformation.py:
```diff
@@ -54,11 +54,17 @@ def transform_request(
         headers: dict,
     ) -> dict:
+        # Create a copy of optional_params without AWS parameters
+        filtered_optional_params = {
```
this is not the responsibility of the transform_request currently. 

Adding this code logic here would make it inconsistent with other bedrock provider flows 

There is no need for this, as the handler is currently taking care of this logic

krrishdholakia commented:
Hi @5S this PR seems unecessary given the handler is already doing this logic. It adds duplicate logic and introduces the scope for bugs. 

I'm closing this for now - if you can share the steps to repro your issue on here https://github.com/BerriAI/litellm/issues/9106, we can go from there
